### PR TITLE
Add Microsoft.AspNetCore.Mvc.Dnx to allow using Mvc with DNX

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @davidfowl \ @pakrym

File: src/Microsoft.AspNetCore.Mvc.Dnx/MvcDnxServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class MvcDnxServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Adds services required for Mvc applications to work with DNX to the specified <paramref name="services"/>.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection" />.</param>
+        /// <returns>A reference to this instance after the operation has completed.</returns>
+        public static IServiceCollection AddMvcDx(this IServiceCollection services)
```
Comment (PinpointTownes) [inline src/Microsoft.AspNetCore.Mvc.Dnx/MvcDnxServiceCollectionExtensions.cs]: nit: `AddMvcDx` -> `AddMvcDnx` :smile:

**Comment by PinpointTownes** [comment]
Exactly the approach I had in mind (cf. https://github.com/openiddict/openiddict-core/issues/64#issuecomment-189321836) :smile: 

:shipit:

**Comment by pakrym** [comment]
:shipit: after method name fix

**Comment by Gillardo** [comment]
Is this going to be available on myget soon?

File: src/Microsoft.AspNetCore.Mvc.Dnx/MvcDnxServiceCollectionExtensions.cs (hunk 1 above)
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Dnx/MvcDnxServiceCollectionExtensions.cs]: Oops.

**Comment by pranavkm** [comment]
Merged in https://github.com/aspnet/Mvc/commit/385c21fbe2af2d61e697b9df826681e6d064d85c

**Comment by pranavkm** [comment]
@Gillardo this should make it to the aspnetcidev feed in a few hours.

**Comment by PinpointTownes** [comment]
Thanks @pranavkm, I really appreciate the effort :clap:

File: samples/MvcSandbox/Startup.cs [hunk 2]
```diff
@@ -14,6 +14,9 @@ public class Startup
         public void ConfigureServices(IServiceCollection services)
         {
             services.AddMvc();
+#if DNX451
+            services.AddMvcDx();
```
Comment (davidfowl) [inline samples/MvcSandbox/Startup.cs]: Dx?

**Comment by davidfowl** [comment]
@pranavkm can you write an announcement?

**Comment by Gillardo** [comment]
Is there a holdup on this being published?  I have done a `dnu restore` and `dotnet restore` and nothing seems to be there?  Also cant see Microsoft.AspNetCore.Mvc.Dnx on myget aspnetcidev repo

**Comment by ciel** [comment]
Yes, I am also having difficulty seeing it. Do we need to do something to get the feed to refresh?

**Comment by PinpointTownes** [comment]
> Do we need to do something to get the feed to refresh?

Nope, we just need to wait for the next CI build.

**Comment by Gillardo** [comment]
How long is that going to take? Day? Weeks?

**Comment by ciel** [comment]
They're called "nightly" builds, so I assume it'll be tomorrow.

File: samples/MvcSandbox/Startup.cs (hunk 2 above)
Comment (pranavkm) [inline samples/MvcSandbox/Startup.cs]: Fixed in the commit.

**Comment by Aequoris** [comment]
If you need this in a hurry like I did, you can take the files out of this pull request and drop them in a folder in your dnx project. They work just as well in there as they would in a seperate Nuget package.

**Comment by ciel** [comment]
The package is live now

**Comment by Gillardo** [comment]
It is now available via myget. I managed to get the files this morning via
`dnu restore`

**Comment by pranavkm** [comment]
Announcement created here - https://github.com/aspnet/Announcements/issues/154. Issue for further questions \ discussion - https://github.com/aspnet/Mvc/issues/4187

**Pull request closed without merging**